pkg/interface/repository/repository.go

```diff
@@ -0,0 +1,224 @@
+package repository
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+
+	"github.com/google/uuid"
+	log "github.com/sirupsen/logrus"
+	"github.com/traPtitech/neoshowcase/pkg/domain"
+	"github.com/traPtitech/neoshowcase/pkg/infrastructure/admindb/models"
+	"github.com/volatiletech/sqlboiler/v4/boil"
+)
+
+//go:generate go run github.com/golang/mock/mockgen@latest -source=$GOFILE -package=mock_$GOPACKAGE -destination=./mock/$GOFILE
+type GitrepositoryRepository interface {
+	RegisterRepository(ctx context.Context, args RegisterRepositoryArgs) (*domain.Repository, error)
+	GetRepositoryByID(ctx context.Context, id string) (*domain.Repository, error)
+	GetRepository(ctx context.Context, args GetRepositoryArgs) (*domain.Repository, error)
+	RegisterProvider(ctx context.Context, args RegisterProviderArgs) (*domain.Provider, error)
+	GetProviderByID(ctx context.Context, id string) (*domain.Provider, error)
+	GetProviderByHost(ctx context.Context, host string) (*domain.Provider, error)
+}
+
+type gitrepositoryRepository struct {
+	db *sql.DB
+}
+
+type RegisterRepositoryArgs struct {
+	RepositoryName  string
+	RepositoryOwner string
+	URL             string
+	ProviderID      string // TODO: providerid型を作る
+}
+
+type RegisterProviderArgs struct {
+	Domain string
+	Secret string
+}
+
+type GetRepositoryArgs struct {
+	ProviderID string
+	Owner      string
+	Name       string
+}
+
+func NewGitrepositoryRepository(db *sql.DB) GitrepositoryRepository {
+	return &gitrepositoryRepository{
+		db: db,
+	}
+}
+
+func (r *gitrepositoryRepository) RegisterRepository(ctx context.Context, args RegisterRepositoryArgs) (*domain.Repository, error) {
+	const errMsg = "failed to RegisterRepository: %w"
+
+	repo, err := models.Repositories(models.RepositoryWhere.URL.EQ(args.URL)).One(ctx, r.db)
+
+	if err != nil && err != sql.ErrNoRows {
+		return nil, fmt.Errorf(errMsg, err)
+	} else if repo != nil {
```

Comment:
```suggestion
	} else if err != nil {
```
網羅的かわかりにくくなるから条件分岐で複数のものを見ないようにしてほしい